tools/package-tools/package-version-stamp.ts

```diff
@@ -0,0 +1,30 @@
+import {writeFileSync} from 'fs';
+import {spawnSync} from 'child_process';
+import {buildConfig} from './build-config';
+
+/**
+ * Inserts the a version stamp, which consists of the current commit SHA and the current
+ * branch name, into the specified package.json file.
+ *
+ * This makes it easy to quickly verify the exact snapshot from which the release originated.
+ */
+export function insertPackageJsonVersionStamp(packageJsonPath: string) {
+  const packageJson = require(packageJsonPath);
+
+  packageJson['_gitCommitStamp'] = getCurrentCommitSha();
```

Comment:
Any reason to prefix with an underscore?

Reply:
No real strong reason. I just wanted to indicate that these properties are not relevant for most developers. I took `underscore` because NPM also does that for their metadata in `package.json`.

![image](https://user-images.githubusercontent.com/4987015/43413927-adda6134-9431-11e8-86db-c551eae771cd.png)

Let me know if you want me to remove it.

Reply:
I would prefer to remove it if `npm` uses it for their own internal properties

Reply:
Sure. I assume the naming itself is fine then? e.g. `gitCommitStamp`.